src/components/DropdownFilter.ts

```diff
@@ -0,0 +1,118 @@
+import { ChangeEvent, Component, OptionHTMLAttributes, ReactElement, createElement } from "react";
+
+import { FilterProps } from "./DropdownFilterContainer";
+import "./ui/DropdownFilter.css";
+
+export interface DropdownFilterProps {
+    defaultFilterIndex: number;
+    enableEmptyFilter: boolean;
+    filters: FilterProps[];
+    handleChange: (FilterProps) => void;
+    placeholder: string;
+}
+
+interface DropdownFilterState {
+    selectedValue: string;
+}
+
+// Added to deal with typings issue of componentClass, needing to pass a className attribute on select options
+export interface DropdownType extends OptionHTMLAttributes<HTMLOptionElement> {
+    value: string;
+    label: string;
+}
+
+type Display = Partial<FilterProps> & DropdownFilterState;
+
+export class DropdownFilter extends Component<DropdownFilterProps, DropdownFilterState> {
+    // Remap prop filters to dropdownfilters
+    private filters: Display[];
+
+    constructor(props: DropdownFilterProps) {
+        super(props);
+
+        // Should have state because select is a controlled component
+        this.state = {
+            selectedValue : this.props.defaultFilterIndex < 0
+                ? "0"
+                : `${this.props.defaultFilterIndex}`
+        };
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    render() {
+        this.filters = this.applyEmptyFilter(this.props.filters);
+        return createElement("select", {
+            className: "form-control",
+            onChange: this.handleOnChange,
+            value: this.state.selectedValue
+        }, this.createOptions());
+    }
+
+    componentDidMount() {
+        // initial state has selectedValue as defaultFilter's index
+        const selectedValue = this.props.defaultFilterIndex < 0 ? "0" : `${this.props.defaultFilterIndex}`;
+        const selectedFilter = this.filters.find(filter => filter.selectedValue === selectedValue);
+        this.props.handleChange(selectedFilter as FilterProps);
+    }
+
+    componentDidUpdate(_prevProps: DropdownFilterProps, prevState: DropdownFilterState) {
+        if (prevState.selectedValue !== this.state.selectedValue) {
+            const selectedFilter = this.filters.find(filter => filter.selectedValue === this.state.selectedValue);
+            this.props.handleChange(selectedFilter as FilterProps);
+        }
+    }
+
+    private createOptions(): Array<ReactElement<{}>> {
+        const options: Array<ReactElement<{}>> = this.filters.map((option, index) => {
+            const optionAttributes: DropdownType = {
+                // Placeholder option is at index 0
+                disabled: (index === 0) ? true : undefined,
```

Comment:
Highly doubt  this will work, correctly when the empty option is disabled in the configuration.

Reply:
It should work because index 0 is the placeholder then the empty option is at index 1.